Argo/Types/Decoded/Monad.swift

```diff
@@ -14,13 +16,15 @@ public func >>- <T, U>(x: Decoded<T>, @noescape f: T -> Decoded<U>) -> Decoded<U
 }
 
 /**
-  flatMap a function over a `Decoded` value (left associative)
+  Conditionally map a function over a `Decoded` value, flattening the result.
 
-  - If the value is a failing case (`.TypeMismatch`, `.MissingKey`), the function will not be evaluated and this will return the value
-  - If the value is `.Success`, the function will be applied to the unwrapped value
+  - If the value is `.Failure`, the function will not be evaluated and this
+    will return `.Failure`.
+  - If the value is `.Success`, the function will be applied to the unwrapped
+    value.
```

Comment:
I suppose the fact that it's `f: T -> Decoded<U>` makes that semi-clear, but I think it's nice to note.